Sources/SwiftProtobuf/Google_Protobuf_FieldMask+Extensions.swift

```diff
@@ -20,6 +20,9 @@ private func ProtoToJSON(name: String) -> String? {
   var jsonPath = String()
   var chars = name.makeIterator()
   while let c = chars.next() {
+    if !c.isASCII {
```

Comment:
maybe `guard` since both usage of this is `if !`?

Reply:
What about `{`, `$`, etc. should those be allowed?  i.e. - should we clamp this down even more?

Reply:
Apparently, the full syntax supported in C++ includes some compressed encodings that I'm not entirely familiar with.  So I'd rather not clamp down too tightly until I better understand that.

`guard` is a good idea.  I need to first find an `isASCII` equivalent that works in Swift 4.2.

Reply:
Reworked the ASCII detection to a form that should be compatible with Swift 4.2

Reply:
I also sketched a next step to filtering the allowed characters more tightly.